feature-libs/product-configurator/rulebased/components/attribute/types/checkbox-list/configurator-attribute-checkbox-list.component.spec.ts

```diff
@@ -106,4 +103,165 @@ describe('ConfigAttributeCheckBoxListComponent', () => {
+        updateType: Configurator.UpdateType.ATTRIBUTE,
+      })
+    );
+  });
+
+  it('should call onHandleAttributeQuantity of event onChangeQuantity', () => {
+    spyOn(component, 'onHandleAttributeQuantity');
+
+    const quantity = { quantity: 2 };
+
+    component.onChangeQuantity(quantity);
+
+    expect(component.onHandleAttributeQuantity).toHaveBeenCalled();
+  });
+
+  it('should call onSelect of event onChangeQuantity', () => {
+    spyOn(component, 'onSelect');
+
+    const quantity = { quantity: 0 };
+
+    component.onChangeQuantity(quantity);
+
+    expect(component.onSelect).toHaveBeenCalled();
+  });
+
+  it('should allow quantity on attribute level when specified so', () => {
+    (component.attribute.dataType =
+      Configurator.DataType.USER_SELECTION_QTY_ATTRIBUTE_LEVEL),
+      expect(component.withQuantityOnAttributeLevel).toBeTruthy();
+  });
+
+  it('should not allow quantity on attribute level when specified as value level', () => {
+    (component.attribute.dataType =
+      Configurator.DataType.USER_SELECTION_QTY_VALUE_LEVEL),
+      expect(component.withQuantityOnAttributeLevel).toBeFalsy();
+  });
+
+  it('should check allowZeroValueQuantity getter', () => {
+    expect(component.allowZeroValueQuantity).toBeTruthy();
+  });
+
+  // HTML
+
+  it('should not display attribute quantity when dataType is no quantity', () => {
+    component.attribute.dataType = Configurator.DataType.USER_SELECTION_NO_QTY;
+
+    fixture.detectChanges();
+
+    CommonConfiguratorTestUtilsService.expectElementNotPresent(
+      expect,
+      htmlElem,
+      'cx-configurator-attribute-quantity'
+    );
+  });
+
+  it('should not display value quantity when dataType is no quantity', () => {
+    component.attribute.dataType = Configurator.DataType.USER_SELECTION_NO_QTY;
+
+    fixture.detectChanges();
+
+    CommonConfiguratorTestUtilsService.expectElementNotPresent(
+      expect,
+      htmlElem,
+      'cx-configurator-attribute-quantity'
+    );
+  });
+
+  it('should display attribute quantity when dataType is with attribute quantity', () => {
+    component.attribute.dataType =
+      Configurator.DataType.USER_SELECTION_QTY_ATTRIBUTE_LEVEL;
+
+    fixture.detectChanges();
+
+    CommonConfiguratorTestUtilsService.expectElementPresent(
+      expect,
+      htmlElem,
+      'cx-configurator-attribute-quantity'
+    );
+  });
+
+  it('should display value quantity when dataType is with value quantity', () => {
+    component.attribute.dataType =
+      Configurator.DataType.USER_SELECTION_QTY_VALUE_LEVEL;
+
+    fixture.detectChanges();
+
+    CommonConfiguratorTestUtilsService.expectElementPresent(
+      expect,
+      htmlElem,
+      'cx-configurator-attribute-quantity'
+    );
+  });
+
+  it('should allow quantity', () => {
+    expect(component.withQuantity).toBe(true);
+  });
+
+  it('should not allow quantity when service is missing ', () => {
+    component['quantityService'] = undefined;
```

Comment:
Can we add some comment here: from my understanding, `quantityService` is not optional dependency. We make it optional now only because breaking change is avoided. So, when we make `quantityService` required in the major release, this test does not make sense any more.

Reply:
picked

Reply:
exactly - but until then the test ensures, that's really optional. When `quantityService `is required, this test has to be removed.

Reply:
done